zerver/tests/test_subs.py

```diff
@@ -5694,6 +5690,15 @@ def test_realm_settings_for_subscribing_other_users(self) -> None:
         )
         self.assert_json_error(result, "Insufficient permission")
 
+        # Moderators, Admins and owners are always full members.
+        self.assertTrue(user_profile.is_provisional_member)
+        do_change_user_role(self.test_user, UserProfile.ROLE_MODERATOR, acting_user=None)
+        self.assertFalse(self.test_user.is_provisional_member)
+        do_change_user_role(self.test_user, UserProfile.ROLE_REALM_ADMINISTRATOR, acting_user=None)
+        self.assertFalse(self.test_user.is_provisional_member)
+        do_change_user_role(self.test_user, UserProfile.ROLE_REALM_OWNER, acting_user=None)
+        self.assertFalse(self.test_user.is_provisional_member)
+
```

Comment:
This test was missing checks for admin and owners and was also placed at the wrong place. This new block would fail without the changes we have made to `is_moderator`